pkg/wallet/wallet.go

```diff
@@ -361,7 +361,32 @@ func (c *Wallet) Verify(options VerificationOption) (bool, error) {
 	}
 }
 
-func (c *Wallet) resolveCredentials(credentials ...CredentialToPresent) ([]*verifiable.Credential, error) {
+// Derive derives a credential and returns response credential.
+//
+//	Args:
+//		- credential to derive (ID of the stored credential, raw credential or credential instance).
+//		- derive options.
+//
+func (c *Wallet) Derive(credential CredentialToDerive,
+	options *DeriveOptions) (*verifiable.Credential, error) {
+	vc, err := c.resolveCredentialToDerive(credential)
+	if err != nil {
+		return nil, fmt.Errorf("failed to resolve request : %w", err)
+	}
+
+	pkFetcher := verifiable.WithPublicKeyFetcher(
+		verifiable.NewVDRKeyResolver(c.ctx.VDRegistry()).PublicKeyFetcher(),
+	)
+
+	derived, err := vc.GenerateBBSSelectiveDisclosure(options.Frame, []byte(options.Nonce), pkFetcher)
```

Comment:
TODO: we should followup with a refactor of `GenerateBBSSelectiveDisclosure` into a `NewCredential` function and WithSelectiveDisclosure(vc, frame, nonce, ...) option.

Reply:
Will create a task for this

Reply:
Created: https://github.com/hyperledger/aries-framework-go/issues/2697